source/configuring-ember/preparing-for-upgrade.md

```diff
@@ -0,0 +1,154 @@
+    { handler: "throw", matchMessage: "Ember.Handlebars.registerHelper is deprecated, please refactor to Ember.Helper.helper." },
+    { handler: "silence", matchMessage: "`lookup` was called on a Registry. The `initializer` API no longer receives a container, and you should use an `instanceInitializer` to look up objects from the container." },
+    { handler: "silence", matchMessage: "Using `Ember.HTMLBars.makeBoundHelper` is deprecated. Please refactor to using `Ember.Helper` or `Ember.Helper.helper`." },
+    { handler: "silence", matchMessage: "Using Ember.HTMLBars._registerHelper is deprecated. Helpers (even dashless ones) are automatically resolved." },
+    { handler: "silence", matchMessage: "In Ember 2.0 service factories must have an `isServiceFactory` property set to true. You registered (unknown mixin) as a service factory. Either add the `isServiceFactory` property to this factory or extend from Ember.Service." },
+    { handler: "silence", matchMessage: "Ember.View is deprecated. Consult the Deprecations Guide for a migration strategy." },
+    { handler: "silence", matchMessage: "Depending on arrays using a dependent key ending with `@each` is deprecated. Please refactor from `Ember.observer('content.@each', function() {});` to `Ember.computed('content.[]', function() {})`." },
+    { handler: "silence", matchMessage: /Accessing 'template' in .* is deprecated. To determine if a block was specified to .* please use '{{#if hasBlock}}' in the components layout./ },
+    { handler: "silence", matchMessage: "A property of <web-directory@view:-outlet::ember1012> was modified inside the didInsertElement hook. You should never change properties on components, services or models during didInsertElement because it causes significant performance degradation." }
+  ]
+};
+```
+
+![Failed Testcase from a Thrown Deprecation](../../images/guides/configuring-ember/preparing-for-upgrade/failed-test-from-deprecation.png)
+
+
+### 3. Fix and Repeat
+After fixing a deprecation and getting your scenarios working again, you might want to leave the deprecation message in the workflow file with the
+throw handler enabled.  This will ensure you haven't missed anything, and ensure no new deprecated calls of that type are introduced to your project.
+Next its just a matter of going down the list, changing the handler, and fixing.
+
+
+In the end your deprecations can be fully turned on as "throw" and you should be able to use your application without error.  At this point you can
+go ahead and up your ember version!  When you upgrade, be sure you remove the deprecation list you've fixed, and update your deprecation handler
+appropriately so that you can begin preparing for the next release.
+
+## Silencing Deprecation Warnings During Compile
+
+You might also notice as you upgrade between releases, that your terminal log begins to stream tons of deprecation warnings during your compile process, making
+it difficult to review your compilation logs.
```

Comment:
Add a sentence explaining that these are completely different template deprecation errors, rather than the same errors that are being suppressed while the app is running.